examples/transmission_main_example.py

```diff
@@ -77,10 +61,63 @@ def main(args):
     axis('off')
     show()
 
+
+def main(args):
+    input_filename = str(args.filename)
+    split_filename = input_filename.split(".")
+    json_filename = split_filename[0]+'.json'
+    try:
+        assert split_filename[1] in ('json','xls','csv','xlsm')
```

Comment:
When you run `python -O` (with optimisations,) assertions are elided from the AST.

What this means is that `assert` is a form of documentation, not a way to check business logic.

Instead of:
```
try:
  assert pred
except AssertionError as e:
  raise pred
```

Do:
```
if not pred:
   raise Exception(...)
```

Use `assert` to communication information to the person reading the code, not to perform critical validations.

Reply:
Furthermore, this logic can be wrapped into the `ArgumentParser`.

Set the `type` to `json_path`:
```
from argparse import ArgumentParser
from pathlib import Path

parser = ArgumentParser()
def json_path(p):
    p = Path(p)
    if p.suffix != '.json':
        parser.error('must be .json file')
    return p
parser.add_argument('filename', type=json_path)

args = parser.parse_args(['abc.txt'])
print(args)

args = parser.parse_args(['abc.json'])
print(args)
```

Reply:
ok, please help here:
I'd like to finish the eqpt part of the code. Maybe you can modify this afterwards?